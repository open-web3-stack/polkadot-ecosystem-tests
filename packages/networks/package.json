{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
