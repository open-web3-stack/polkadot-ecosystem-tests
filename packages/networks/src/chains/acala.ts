import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
