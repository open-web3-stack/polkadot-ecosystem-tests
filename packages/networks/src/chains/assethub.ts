import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
