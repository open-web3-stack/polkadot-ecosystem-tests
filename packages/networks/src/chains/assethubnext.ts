import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
