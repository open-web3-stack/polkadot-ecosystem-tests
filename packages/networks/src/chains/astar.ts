import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
