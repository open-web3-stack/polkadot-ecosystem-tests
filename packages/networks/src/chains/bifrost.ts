import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
