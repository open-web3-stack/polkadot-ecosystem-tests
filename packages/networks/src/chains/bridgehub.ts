import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
