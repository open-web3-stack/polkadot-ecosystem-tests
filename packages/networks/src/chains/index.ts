export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
