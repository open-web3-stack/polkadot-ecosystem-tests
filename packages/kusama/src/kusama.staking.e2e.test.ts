import { kusama } from '@e2e-test/networks/chains'

import { fullStakingTests, registerTestTree } from '@e2e-test/shared'

registerTestTree(fullStakingTests(kusama, { testSuiteName: 'Kusama Staking', addressEncoding: 2 }))
