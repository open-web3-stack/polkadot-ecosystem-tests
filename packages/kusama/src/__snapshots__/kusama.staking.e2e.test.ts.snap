// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kusama Staking > cancel deferred slash with bad origin > cancel deferred slash events with bad origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 77000)",
          "refTime": "(rounded 1600000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 2`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 3`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 4`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 5`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 6`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other bad origin events 7`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x15000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > chill other > chill other events 1`] = `
[
  {
    "data": {
      "stash": "FoQJpPyadYccjavVdTWxpxU7rUEaYhfLCPwXgkfD6Zat9QP",
    },
    "method": "Chilled",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > force apply validator commission > force apply min commission events 1`] = `
[
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 11000)",
          "refTime": "(rounded 640000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > modify validator count > increase validator count bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 7200)",
          "refTime": "(rounded 570000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > modify validator count > scale validator count bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 7200)",
          "refTime": "(rounded 570000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > modify validator count > set validator count bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 7200)",
          "refTime": "(rounded 570000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > set invulnerables with bad origin > events when setting invulnerables with bad staking admin origin 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": "BadOrigin",
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Kusama Staking > set invulnerables with bad origin > set invulnerables events with bad signed origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 7200)",
          "refTime": "(rounded 570000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > set minimum validator commission > set staking configs bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 7200)",
          "refTime": "(rounded 570000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > set staking configs > set staking configs bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 7200)",
          "refTime": "(rounded 1200000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > set staking configs > set staking configs event 1`] = `
{
  "event": {
    "data": [
      {
        "class": "Mandatory",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 1500)",
          "refTime": "(rounded 290000000)",
        },
      },
    ],
    "index": "0x0000",
  },
  "phase": {
    "applyExtrinsic": 0,
  },
  "topics": [],
}
`;

exports[`Kusama Staking > staking lifecycle > chill events 1`] = `
[
  {
    "data": {
      "stash": "FQ3JxEebQJY23NumtMUU1k7YJrbyK6L8j4uK5VwKpeVqFnh",
    },
    "method": "Chilled",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > events when attempting to nominate a blocked validator 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x14000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 21000)",
          "refTime": "(rounded 1600000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > kick events 1`] = `
[
  {
    "data": {
      "nominator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "stash": "FQ3JxEebQJY23NumtMUU1k7YJrbyK6L8j4uK5VwKpeVqFnh",
    },
    "method": "Kicked",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > nominate events 1`] = `[]`;

exports[`Kusama Staking > staking lifecycle > nominator bond events 1`] = `
[
  {
    "data": {
      "amount": 100000000000000,
      "stash": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > nominator bond extra events 1`] = `
[
  {
    "data": {
      "amount": 100000000000000,
      "stash": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > unbond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "FQ3JxEebQJY23NumtMUU1k7YJrbyK6L8j4uK5VwKpeVqFnh",
    },
    "method": "Unbonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validate (blocked) events 1`] = `
[
  {
    "data": {
      "prefs": {
        "blocked": true,
        "commission": "10.00%",
      },
      "stash": "FQ3JxEebQJY23NumtMUU1k7YJrbyK6L8j4uK5VwKpeVqFnh",
    },
    "method": "ValidatorPrefsSet",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validator 0 bond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "FQ3JxEebQJY23NumtMUU1k7YJrbyK6L8j4uK5VwKpeVqFnh",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validator 0 validate events 1`] = `
[
  {
    "data": {
      "prefs": {
        "blocked": false,
        "commission": "10.00%",
      },
      "stash": "FQ3JxEebQJY23NumtMUU1k7YJrbyK6L8j4uK5VwKpeVqFnh",
    },
    "method": "ValidatorPrefsSet",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validator 1 bond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "GtgvkKoi3CxJk2mP5ZzuJ7Mj8ipiy4Sf6T4wtt89W8m8FNa",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validator 1 validate events 1`] = `
[
  {
    "data": {
      "prefs": {
        "blocked": false,
        "commission": "10.00%",
      },
      "stash": "GtgvkKoi3CxJk2mP5ZzuJ7Mj8ipiy4Sf6T4wtt89W8m8FNa",
    },
    "method": "ValidatorPrefsSet",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validator 2 bond events 1`] = `
[
  {
    "data": {
      "amount": 50000000000000,
      "stash": "ErRmn1hykJ1ppjMoNfQyRYA9c3j7y3YZnKQsCecD1k9D42B",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > staking lifecycle > validator 2 validate events 1`] = `
[
  {
    "data": {
      "prefs": {
        "blocked": false,
        "commission": "10.00%",
      },
      "stash": "ErRmn1hykJ1ppjMoNfQyRYA9c3j7y3YZnKQsCecD1k9D42B",
    },
    "method": "ValidatorPrefsSet",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > test fast unstake > nominate events 1`] = `[]`;

exports[`Kusama Staking > test fast unstake > nominator bond events 1`] = `
[
  {
    "data": {
      "amount": 100000000000000,
      "stash": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "Bonded",
    "section": "staking",
  },
]
`;

exports[`Kusama Staking > test fast unstake > register fast unstake events 1`] = `[]`;

exports[`Kusama Staking > test force unstaking of nominator > force unstake bad origin events 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 13000)",
          "refTime": "(rounded 2200000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > trying to become a validator with no bonded funds fails > events when attempting to validate with no bonded funds 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x00000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 12000)",
          "refTime": "(rounded 1300000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > trying to nominate with no bonded funds fails > events when attempting to nominate with no bonded funds 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x00000000",
          "index": 6,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 16000)",
          "refTime": "(rounded 1500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Staking > unapplied slash > balances slash events 1`] = `
[
  {
    "data": {
      "amount": 5000000000000,
      "who": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "Slashed",
    "section": "balances",
  },
  {
    "data": {
      "amount": 10000000000000,
      "who": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "Slashed",
    "section": "balances",
  },
  {
    "data": {
      "amount": 10000000000000,
      "who": "D2oHoVDFJGy7iDYcY7fbm82CuNycWaSxCGpsQtLqkrE9JKf",
    },
    "method": "Slashed",
    "section": "balances",
  },
]
`;

exports[`Kusama Staking > unapplied slash > staking slash events 1`] = `
[
  {
    "data": {
      "amount": 5000000000000,
      "staker": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "Slashed",
    "section": "staking",
  },
  {
    "data": {
      "amount": 10000000000000,
      "staker": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "Slashed",
    "section": "staking",
  },
  {
    "data": {
      "amount": 10000000000000,
      "staker": "D2oHoVDFJGy7iDYcY7fbm82CuNycWaSxCGpsQtLqkrE9JKf",
    },
    "method": "Slashed",
    "section": "staking",
  },
]
`;
